mantidproject/mantid
# Muon Background Subtraction - Minus the background from the counts

**Description of work.**
This PR applies the calculated background to the Counts data using the `Minus` algorithm. The Asymmetry data is then updated as it uses the Counts data to be calculated.

**To test:**
1. Open Muon Analysis and load PSI data
2. Go Corrections tab. Select `Auto` background corrections
3. Try the following combinations. Check that the loaded data is corrected accordingly:
 - `Auto` with `Flat Background` between `5.0` and `9.8` should show `Correction skipped`
 - `Auto` with `Flat Background + Exp Decay` between `5.0` and `9.8` should show `Correction success`
 - `Auto` with `Flat Background` between `0` and `0.14` should show `Correction success` for timezero set to 0.0


Fixes #31816

*This does not require release notes* because **the full background corrections feature is not yet complete.**. Release notes will be added in a future PR.

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
